src/test/java/uk/gov/hmcts/divorce/document/content/provider/ApplicationTemplateDataProviderTest.java

```diff
@@ -64,12 +64,62 @@ void shouldReturnEmptyListOfJurisdictionsIfNoJurisdictionConnectsSelected() {
     }
 
     @Test
-    void shouldReturnEmptyListOfJurisdictionsIfJurisdictionConnectionsIsNull() {
+    void shouldReturnEmptyListOfJurisdictionsForJointIfJurisdictionConnectionsIsNull() {
 
         final var application = Application.builder().build();
 
         final var result = applicationTemplateDataProvider.deriveJointJurisdictionList(application);
 
         assertThat(result).isEmpty();
     }
+
+    @Test
+    void shouldReturnListOfJurisdictionsForSoleIfAllSelected() throws Exception {
+
+        final var application = Application.builder().build();
+        application.getJurisdiction().setConnections(Set.of(
+            APP_1_APP_2_RESIDENT,
+            APP_1_APP_2_LAST_RESIDENT,
+            APP_1_RESIDENT_TWELVE_MONTHS,
+            APP_1_RESIDENT_SIX_MONTHS,
+            APP_1_APP_2_DOMICILED,
+            RESIDUAL_JURISDICTION));
+
+        final var result = applicationTemplateDataProvider.deriveSoleJurisdictionList(application);
+
+        assertThat(result).containsExactly(
+            new Connection("the applicant and respondent are habitually resident in England and Wales"),
+            new Connection("the applicant and respondent were last habitually resident in England and Wales "
+                + "and one of them still resides there"),
+            new Connection("the applicant is habitually resident in England and Wales and has resided there "
+                + "for at least a year immediately prior to the presentation of the application"),
+            new Connection("the applicant is domiciled and habitually resident in England and Wales and has "
+                + "resided there for at least six months immediately prior to the application"),
+            new Connection("the applicant and respondent are both domiciled in England and Wales"),
+            new Connection("the applicant and respondent registered as civil partners of each other in England "
+                + "or Wales or, in the case of a same sex couple, married each other under the law of England and Wales "
+                + "and it would be in the interests of justice for the court to assume jurisdiction in this case")
+        );
+    }
+
+    @Test
+    void shouldReturnEmptyListOfJurisdictionsForSoleIfNoJurisdictionConnectsSelected() {
```

Comment:
I think if juridiction list is empty we should throw an exception as we cannot apply for divorce without jurisdiction selection. So ideally template provider should throw an error

Reply:
Added exception 
https://github.com/hmcts/nfdiv-case-api/pull/747/commits/408187f6680fd66c7b637bf28f8a54ac43c8572a